sdk/ai/azopenai/custom_client.go

```diff
@@ -277,25 +271,66 @@ type clientData struct {
 	azure    bool
 }
 
-func getDeployment[T ChatCompletionsOptions | CompletionsOptions | EmbeddingsOptions | ImageGenerationOptions | *getAudioTranscriptionInternalOptions | *getAudioTranslationInternalOptions](v T) string {
+func getDeployment[T AudioTranscriptionOptions | AudioTranslationOptions | ChatCompletionsOptions | CompletionsOptions | EmbeddingsOptions | *getAudioTranscriptionInternalOptions | *getAudioTranslationInternalOptions | ImageGenerationOptions](v T) string {
+	var p *string
+
 	switch a := any(v).(type) {
+	case AudioTranscriptionOptions:
+		p = a.DeploymentName
+	case AudioTranslationOptions:
+		p = a.DeploymentName
 	case ChatCompletionsOptions:
-		return a.Deployment
+		p = a.DeploymentName
 	case CompletionsOptions:
-		return a.Deployment
+		p = a.DeploymentName
 	case EmbeddingsOptions:
-		return a.Deployment
-	case ImageGenerationOptions:
-		return ""
+		p = a.DeploymentName
 	case *getAudioTranscriptionInternalOptions:
-		return *a.Model
+		p = a.Model
 	case *getAudioTranslationInternalOptions:
-		return *a.Model
-	default:
-		return ""
+		p = a.Model
+	case ImageGenerationOptions:
+		p = a.DeploymentName
+	}
+
+	if p != nil {
+		return *p
 	}
+
+	return ""
 }
 
 func hasAzureExtensions(body ChatCompletionsOptions) bool {
-	return body.AzureExtensionsOptions != nil && len(body.AzureExtensionsOptions.Extensions) > 0
+	return body.AzureExtensionsOptions != nil && len(body.AzureExtensionsOptions) > 0
+}
+
+// ChatRequestUserMessageContent contains the user prompt - either as a single string
+// or as a []ChatCompletionRequestMessageContentPart, enabling images and text as input.
+//
+// NOTE: This should be created using [azopenai.NewChatRequestUserMessageContent]
+type ChatRequestUserMessageContent struct {
+	str   *string
```

Comment:
Since these are just inputs to `json.Marshal` couldn't they be combined `any` field? Type info doesn't seem to be needed at this point.

Reply:
Can be left as-is for now. If in the future more fields are added then this will need to be refactored.

Reply:
No, no, your idea makes sense. I just misunderstood it.